python/GafferUI/GraphBookmarksUI.py

```diff
@@ -147,3 +181,66 @@ def __connect( inPlug, outPlug ) :
 
 	with Gaffer.UndoScope( inPlug.ancestor( Gaffer.ScriptNode ) ) :
 		inPlug.setInput( outPlug )
+
+def __editBookmark( editorWeakRef, bookmark ) :
+
+	editor = editorWeakRef()
+
+	if isinstance( editor, GafferUI.GraphEditor ) :
+		editor.graphGadget().setRoot( bookmark.parent() )
+		editor.frame( [ bookmark ] )
+	else :
+		editor.setNodeSet( Gaffer.StandardSet( [ bookmark ] ) )
+
+def __findableBookmarks( editor ) :
+
+	# Prefer this over manual recursion through the graph because it is _much_
+	# faster.
+	bookmarks = Gaffer.Metadata.nodesWithMetadata( editor.scriptNode(), "bookmarked", instanceOnly = True )
+	return [ b for b in bookmarks if b.ancestor( Gaffer.Reference ) is None ]
```

Comment:
I think you need to add `if Gaffer.Metadata.value( x, "bookmarked" )` here, otherwise previously bookmarked nodes remain in the list.

Reply:
Either that or update the un-bookmarking to remove the metadata entirely (and provide compatibility for old files that have serialized it false).

Reply:
Good catch - I'd just assumed that `setBookmarked( False )` cleaned things up. I've made that the case in d8498c4, and dealt with loading of old files in  3b15d2f .